### PR TITLE
LPS-189049 Turn dataset app data model into system objects

### PR DESCRIPTION
**Pull request by dsanz**
In this PR we are changing the way we create the object definitions supporting dataset manager persistence model. We are using System object definitions now.

This has implications in:
* The API we use to create the definitions
* The naming of the headless endpoints (context path)
* The naming of the object definitions themselves, which affects how we calculate the class name in various places

Other notes:
* There is no upgrade process as application is under FF
* We are not renaming the object definitions to the UI/user friendly abstractions and therefore the endpoint names don't change (just the context path does)

### CODE REVIEW

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/definition/util/ObjectDefinitionUtil.java [hunk 1]
```diff
@@ -60,6 +60,18 @@ public static boolean isAllowedModifiableSystemObjectDefinitionName(
 			"APISort", "/headless-builder/sorts"
 		).put(
 			"Bookmark", "/bookmarks"
+		).put(
+			"FDSDateFilter", "/data-set-manager/datefilters"
+		).put(
+			"FDSDynamicFilter", "/data-set-manager/dynamicfilters"
```
Comment (markocikos) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/definition/util/ObjectDefinitionUtil.java:66]: maybe `/date-filters` and `/dynamic-filters`?

**Review by markocikos — APPROVED** [review]
@dsanz The code looks good! Minor question about naming inline.

**Comment by dsanz** [comment]
ci:test:sf

**Comment by dsanz** [comment]
ci:test:relevant

**Comment by dsanz** [comment]
ci:test:stable

**Comment by dsanz** [comment]
ci:test:stable

**Comment by dsanz** [comment]
ci:test:relevant

**Comment by dsanz** [comment]
ci:test:sf

**Comment by dsanz** [comment]
ci:sf

**Comment by dsanz** [comment]
ci:test:sf

**Comment by dsanz** [comment]
ci:reevaluate:1257401_1696

**Comment by dsanz** [comment]
ci:reevaluate:1347401_4344

**Comment by mateusralv** [comment]
I'm taking a look 🕵🏽

**Comment by mateusralv** [comment]
ci:test:frontend-data-set

**Comment by mateusralv** [comment]
ci:help

**Comment by dsanz** [comment]
Looks like the calls to addSystemObjectDefinition require some revamp. Taking a look

**Comment by dsanz** [comment]
ci:test:sf
ci:test:relevant
ci:test:stable

**Comment by dsanz** [comment]
ci:test:stable

**Comment by dsanz** [comment]
ci:test:relevant

**Comment by dsanz** [comment]
ci:test:relevant

**Comment by dsanz** [comment]
ci:test:sf

**Comment by dsanz** [comment]
ci:test:stable

**Comment by dsanz** [comment]
ci:test:stable

**Comment by dsanz** [comment]
ci:test:relevant

**Comment by dsanz** [comment]
ci:test:relevant

**Comment by dsanz** [comment]
ci:test:sf

**Comment by dsanz** [comment]
ci:test:stable

**Comment by dsanz** [comment]
ci:test:sf

**Comment by dsanz** [comment]
ci:test:stable

**Comment by dsanz** [comment]
ci:test:relevant

**Comment by dsanz** [comment]
ci:test:relevant

**Comment by mateusralv** [comment]
Hi @dsanz the failures are not related! 

[Testray Comparison](https://testray.liferay.com/home/-/testray/runs/compare?formDate=1688386574541&testrayRunIdA=2367727680&testrayRunIdB=2367948451)

**Comment by dsanz** [comment]
see https://github.com/liferay-frontend/liferay-portal/pull/3518

**Pull request closed without merging**